pkg/ioutils/fswriters.go

```diff
@@ -17,6 +17,20 @@ type AtomicFileWriterOptions struct {
 	// On successful return from Close() this is set to the mtime of the
 	// newly written file.
 	ModTime time.Time
+	// Specifies whether Commit() must be explicitly called to write state
+	// to the destination. This allows an application to preserve the original
+	// file when an error occurs during processing (and not just during write)
+	// The default is false, which will auto-commit on Close
+	ExplicitCommit bool
```

Comment:
I think I'm missing a point and wonder why `ExplicitCommit` and `Commit()` are needed.  I somehow expected changes to `Close()` only.  @n1hility, can you elaborate on that?

Reply:
Thanks, makes sense :+1:

Reply:
Great summary, I couldn't have said it better. I would have preferred to make a hard move and make this commit() approach the only option instead of a flag like in my PR, since the previous auto-commit behavior is likely to cause problems in most usage patterns. However, it would be compatibility breaking, and I wasnt sure how much usage of this API there is in the wild.

Reply:
I think that's a great way of keeping the API stable, @n1hility.  Thanks!